### ISSUE TITLE
Doctrine not committing anyting

### ISSUE THREAD

**Issue by Yunyun548** [doctrine/DoctrineBundle#543]
Typically when I'm trying to insert a new line into an existing table, doctrine isn't committing anything.

```
doctrine.DEBUG: "START TRANSACTION" [] []
doctrine.DEBUG: INSERT INTO FAQ (question, answer, theme, user_id) VALUES (?, ?, ?, ?) {"1":"Doit-on imprimer tous les  [...]","2":"Il vaudrait mieux, même s' [...]","3":"Système qualité","4":null} []

```
The following line is never present in my log file.

`"COMMIT" [] []`

I use 

>  "doctrine/orm": "^2.5.4"
>  "doctrine/doctrine-bundle": "~1.6.2"

I encounter this problem for all INSERT operations

**Comment by mikeSimonson** [comment]
Are you calling ```$em->flush``` ?

**Comment by Yunyun548** [comment]
Yes, I am importing data from another database

For each entity my code looks like this

        ```
        $em = $this->getContainer()->get('doctrine.orm.default_entity_manager');
        $emOld = $this->getContainer()->get('doctrine.orm.olddatabase_entity_manager');

        // Migration FAQ
        $count = 0;
        $oldFaqs = $emOld->getRepository('ImportBundle:Faq')->findAll();
        foreach ($oldFaqs as $of) {
            $newFaq = new Faq();
            $newFaq->setTheme($of->getIdfaqtheme()->getLibelle());
            $newFaq->setQuestion($of->getQuestion());
            $newFaq->setAnswer($of->getReponse());
            $em->persist($newFaq);
            ++$count;
        }
        $em->flush();
        $output->writeln($count . " FAQ created.");
```

**Comment by mikeSimonson** [comment]
Are you sure that the script ends and that it's not killed because it use too much memory ?
That kind of migration should be done using sql directly anyway. You are copying your whole table in memory to put it back in the db for no reason. It's going to be much slower than pure sql.

**Comment by Yunyun548** [comment]
> For example, if my OAuthBundle tries to save an access_token, the access_token isn't saved, breaking the whole application (:

I have this error on every insert statement my application is trying to do, this was just an example.

**Comment by eugene-matvejev** [comment]
@Yunyun548 

init transaction at the start, commit whenever you dispatch response/command output (if it been succesful for example)

**Comment by Yunyun548** [comment]
As I said in my previous post, if I do the following :

 ```
            try {
                        $em->getConnection()->beginTransaction();

                        // Migration FAQ
                        $count = 0;
                        $oldFaqs = $emOld->getRepository('ImportBundle:Faq')->findAll();
                        foreach ($oldFaqs as $of) {
                            $newFaq = new Faq();
                            $newFaq->setTheme($of->getIdfaqtheme()->getLibelle());
                            $newFaq->setQuestion($of->getQuestion());
                            $newFaq->setAnswer($of->getReponse());
                            $em->persist($newFaq);
                            ++$count;
                    }
                    $em->getConnection()->commit();
        } catch (\Exception $e) {
                   $em->getConnection()->rollback();
                   throw $e;
        }
```

But I would really like to understand why `$em->flush()` isn't working this I can't update all the application...

**Comment by Yunyun548** [comment]
As I said in my last post, if I try the following :

```
try {
            $em->getConnection()->beginTransaction();

            // Migration FAQ
            $count = 0;
            $oldFaqs = $emOld->getRepository('ImportBundle:Faq')->findAll();
            foreach ($oldFaqs as $of) {
                $newFaq = new Faq();
                $newFaq->setTheme($of->getIdfaqtheme()->getLibelle());
                $newFaq->setQuestion($of->getQuestion());
                $newFaq->setAnswer($of->getReponse());
                $em->persist($newFaq);
                ++$count;
            }
            $em->getConnection()->commit();
        } catch (\Exception $e) {
            $em->getConnection()->rollback();
            throw $e;
        }
```

my entities are persisted. However I would really like to understand why `$em->flush()` isn't working since I really can't afford to update my whole application.

**Comment by mikeSimonson** [comment]
Well probably an issue with your configuration can you paste it ?

**Comment by Yunyun548** [comment]
I tried to include all the relevant parts

However, I want to add that all the application has been working for weeks and it suddenly stopped for no apparent reason (no composer update/configuration changes) on every environment

```
imports:
    - { resource: parameters.yml }
    - { resource: security.yml }
    - { resource: services.yml }
    - { resource: "@RestBundle/Resources/config/services.yml" }
    - { resource: "@AdminBundle/Resources/config/services.yml" }
    - { resource: easyadmin/general.yml }

# Put parameters here that don't need to change on each machine where the app is deployed
# http://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration
parameters:
    default_locale:  "fr"
    jms_serializer.doctrine_proxy_subscriber.class: RestBundle\Serializer\DoctrineProxySubscriber

framework:
    #esi:             ~
    translator: { fallbacks: [ "fr" ] }
    secret:          "%secret%"
    router:
        resource: "%kernel.root_dir%/config/routing.yml"
        strict_requirements: ~
    form:            ~
    csrf_protection: ~
    validation:      { enable_annotations: true }
    templating:
        engines: ['twig']
    default_locale:  "fr"
    trusted_hosts:   ~
    trusted_proxies: ~
    session:
        handler_id:  ~
    fragments:       ~
    http_method_override: true

# Twig Configuration
twig:
    debug:            "%kernel.debug%"
    strict_variables: "%kernel.debug%"

# Doctrine Configuration
doctrine:
    dbal:
        default_connection: default
        connections:
            default:
                driver:   '%database_driver%'
                host:     '%database_host%'
                port:     '%database_port%'
                dbname:   '%database_name%'
                user:     '%database_user%'
                password: '%database_password%'
                charset:  UTF8
            olddatabase:
                driver:   '%database_driver%'
                host:     '%database_host%'
                port:     '%database_port%'
                dbname:   '%database_name2%'
                user:     '%database_user%'
                password: '%database_password%'
                charset:  UTF8

    orm:
        auto_generate_proxy_classes: "%kernel.debug%"
        # naming_strategy: doctrine.orm.naming_strategy.underscore
        # auto_mapping: true
        default_entity_manager: default
        entity_managers:
            default:
                connection: default
                auto_mapping: true
            olddatabase:
                connection: olddatabase
                mappings:
                    ImportBundle:  ~

# Migration Configuration
doctrine_migrations:
    dir_name: "%kernel.root_dir%/DoctrineMigrations"
    namespace: Application\Migrations
    table_name: migration_versions
    name: Application Migrations

# Serialization
jms_serializer:
    metadata:
        auto_detection: true
        directories:
            RestBundle:
                namespace_prefix: "\\RestBundle"
                path: "@RestBundle/Resources/config/serializer"

# Swiftmailer Configuration
swiftmailer:
    transport: "%mailer_transport%"
    host:      "%mailer_host%"
    port:      "%mailer_port%"
    username:  "%mailer_user%"
    password:  "%mailer_password%"
    spool:     { type: memory }

# Files Uploads
vich_uploader:
    db_driver: orm
    mappings:
        documentation_pdf:
            uri_prefix:         /files/documentation
            upload_destination: %kernel.root_dir%/../web/files/documentation
        # inject_on_load:     false
        # delete_on_update:   true
        # delete_on_remove:   true

# Api Configuration
fos_rest:
    routing_loader:
        default_format: json
        include_format: true
    view:
        view_response_listener: force

    format_listener:
        enabled: true
        rules:
            - { path: '^/espace-administration', methods: ['GET', 'POST'], priorities: ['html'], fallback_format: 'html', prefer_extension: false }
            - { path: '^/', priorities: [ 'html', '*/*'], fallback_format: html, prefer_extension: true }

# Api Doc Configuration
nelmio_api_doc:
    name: 'Reseau RVP API documentation'

# FOS User
fos_user:
    db_driver: orm
    firewall_name: api
    user_class: RestBundle\Entity\User

# FOS OAuthServer
fos_oauth_server:
    db_driver:           orm
    client_class:        RestBundle\Entity\OAuthClient
    access_token_class:  RestBundle\Entity\OAuthToken
    refresh_token_class: RestBundle\Entity\OAuthRefreshToken
    auth_code_class:     RestBundle\Entity\OAuthCode
    service:
        user_provider: fos_user.user_manager

# Nelmio Cors Bundle
nelmio_cors:
   defaults:
       allow_credentials: true
       allow_origin: []
       allow_headers: []
       allow_methods: []
       expose_headers: []
       max_age: 0
       hosts: []
   paths:
    '^/':
        allow_origin: ['*']
        allow_headers: ['origin', 'content-type', 'Authorization']
        allow_methods: ['POST', 'PUT', 'GET', 'DELETE','OPTIONS']
        max_age: 3600
```

**Comment by mikeSimonson** [comment]
In your post with the migration code in try catch there is no ```$em->flush()``` statement.

Also your orm configuration the default one has the key auto mapping to true, you can't use auto mapping when you work with multiple entity manager.

Well something obviously changed in your app it's really hard for me to know what it is but with git you should be able to see what changed.

**Comment by Yunyun548** [comment]
In my configuration isn't the mapping only enabled for one of the entity managers ? The default one ? I remember following this thread when I did that (https://github.com/doctrine/DoctrineBundle/issues/60).

I agree, I forgot to paste the `$em->flush();` line

I have tried with it and haven't seen any different result.

I have checked on git and nothing but controllers and unit tests have been modified since this issue appeared. Is there somewhere I should try to log to get more information about what's happening ? A verbose mode for doctrine ?

**Comment by mikeSimonson** [comment]
Git stash the change and try again if it works then you need to find what it is in your stashed change that broke it.

**Comment by Yunyun548** [comment]
Maybe I explained myself wrong, it was working perfectly on our 3 developers computers and dev server on Friday, on each of our repository branches

Today it wasn't working on any environment. I tried on my personal computer which have a totally different setup and experienced the same problem

**Comment by mikeSimonson** [comment]
The only explaination that make sense is that you depend on something and that you are not aware of it. That untracked dependency changed in some way that broke your setup. I have no idea what it could be. You should try to use xdebug to get a clue.

**Comment by Yunyun548** [comment]
So, I have run some tests.

If I recreated my application (copy-paste the code and config in a new symfony project and run a composer install since it wasn't working with a git clone) everything is working as expected.

I used a bit of XDebug to see what was going on in my command. And didn't saw anything surprising, both entity managers connections where open, both set to auto-commit, all the entities in my context were stuff that should be here. The only calls made before the command started were to scripts needed to start a Symfony command

1. main
2. Symfony\Component\Console\Application->run
3. Symfony\Bundle\FrameworkBundle\ConsoleApplication->doRun
4. Symfony\Component\Console\Application->doRun
5. Symfony\Component\Console\Application->doRunCommand
6. Symfony\Component\Console\Command\Command->run
7. MyCommand->execute

So it looks like my problem is solved, however I have absolutely no idea why and lost quite a bit of time :angry:, and that's kinda frustrating.
